File: app/src/main/java/io/neoterm/setup/connection/OfflineUriConnection.java
package io.neoterm.setup.connection;

import android.content.Context;
import android.net.Uri;

import java.io.IOException;
import java.io.InputStream;

/**
 * @author kiva
 */

public class OfflineUriConnection extends OfflineConnection {
    private final Context context;
    private final Uri uri;

    public OfflineUriConnection(Context context, Uri uri) {
        this.context = context;
        this.uri = uri;
    }

    @Override
    protected InputStream openInputStream() throws IOException {
        return context.getContentResolver().openInputStream(uri);
    }
}

File: app/src/main/java/io/neoterm/setup/connections/OfflineUriConnection.kt
package io.neoterm.setup.connections

import android.content.Context
import android.net.Uri

import java.io.IOException
import java.io.InputStream

/**
 * @author kiva
 */

open class OfflineUriConnection(private val context: Context, private val uri: Uri) : OfflineConnection() {

    @Throws(IOException::class)
    override fun openInputStream(): InputStream {
        return context.contentResolver.openInputStream(uri)
    }
}
